cram/cram_encode.c

```diff
@@ -1442,8 +1723,20 @@ int cram_encode_container(cram_fd *fd, cram_container *c) {
         if (fd->embed_ref <= 1) {
             char *ref = cram_get_ref(fd, bam_ref(b), 1, 0);
             if (!ref && bam_ref(b) >= 0) {
-                hts_log_error("Failed to load reference #%d", bam_ref(b));
-                return -1;
+                if (c->multi_seq || fd->embed_ref == 0 || !c->pos_sorted) {
+                    hts_log_error("Failed to load reference #%d", bam_ref(b));
+                    return -1;
+                }
+                hts_log_warning("Failed to load reference #%d", bam_ref(b));
+                hts_log_warning("Enabling embed_ref=2 mode to auto-generate"
+                                " reference");
+                if (fd->embed_ref <= 0)
+                    hts_log_warning("NOTE: the CRAM file will be bigger than"
+                                    " using an external reference");
+                pthread_mutex_lock(&fd->ref_lock);
```

Comment:
Good idea for putting into a local variable, as is already done just about with `nref`.  Regarding elsewhere (such as `process_read`) I did think about locks but the various automated tools seemed to think it was impossible to trigger data races there.

I'll explore again with more tools.  It's possible it can still happen if we fail to find a reference part way through a file instead of only at the first one (which essentially means it hasn't yet dispatched a single thread job, hence impossible to trigger).  So I'll stress test more avenues.